src/common-utils/devcontainer-feature.json

```diff
@@ -64,6 +64,18 @@
             "type": "boolean",
             "default": false,
             "description": "Add packages from non-free Debian repository? (Debian only)"
+        },
+        "allowShellHistory": {
+            "type": "boolean",
+            "default": false,
+            "description": "Preserve shell history across dev container instances? (Currently supports bash, zsh, and fish)"
+        }
+    },
+    "mounts": [
+        {
+            "source": "${devcontainerId}-shellhistory",
```

Comment:
This will add a volume for each container which will make volumes harder to manage for the user. Maybe you could use a single volume `devcontainers` at `/devcontainers` and in that create a folder `/devcontainers/shellHistory/${devcontainerId}` for each dev container.

Reply:
have added logic such that uuidgen utility is used for generating random devcontainer id for a container, intent is to embed it inside the folder structure like this :
/devcontainers/$(uuidgen)/shellHistory for a symlink to the ~/.bash_history being kept here. Logic is working for retention of shell history, keeping it as draft so you can check for bash shell at the moment, similarly can be implemented for zsh as well.